network/framework/src/noise/handshake.rs

```diff
@@ -165,13 +108,38 @@ impl NoiseUpgrader {
     pub fn new(
         network_context: NetworkContext,
         key: x25519::PrivateKey,
-        auth_mode: HandshakeAuthMode,
+        peers_and_metadata: Arc<PeersAndMetadata>,
+        mutual_auth: bool,
     ) -> Self {
+        let anti_replay_timestamps = if mutual_auth {
+            Some(RwLock::new(AntiReplayTimestamps::default()))
+        } else {
+            None
+        };
         Self {
             network_context,
             noise_config: noise::NoiseConfig::new(key),
-            auth_mode,
+            peers_and_metadata,
+            anti_replay_timestamps,
+        }
+    }
+
+    pub fn is_mutual_mode(&self) -> bool {
+        self.anti_replay_timestamps.is_some()
+    }
+
+    pub fn is_trusted_peer(&self, network_id: &NetworkId, peer_id: &PeerId) -> Option<Peer> {
```

Comment:
nit: `is_trusted_peer` makes me want to expect a `bool` 😄

Reply:
renaming to `peer_config_if_trusted()`